### PR TITLE
[BUGFIX] Ensure array value is set when accessing

### PR DESCRIPTION
**Pull request by magicHatOfTYPO3**
Before accessing the dataHandler array ensure the requested key is available and skip, if this is not the case. 

Fixes: #3269

### CODE REVIEW

File: Classes/IndexQueue/RecordMonitor.php [hunk 1]
```diff
@@ -130,7 +130,11 @@ public function processDatamap_afterDatabaseOperations(
         }
 
         if ($status === 'new' && !MathUtility::canBeInterpretedAsInteger($recordUid)) {
-            $recordUid = $tceMain->substNEWwithIDs[$recordUid];
+            if (isset($this->dh->substNEWwithIDs[$recordUid])) {
+                $recordUid = $this->dh->substNEWwithIDs[$recordUid];
```
[the remark this answers is not in the document]
Comment (magicHatOfTYPO3) [reply Classes/IndexQueue/RecordMonitor.php]: Good question, seems like a copy'n'paste error. Sorry. `$tceMain->substNEWwithIDs` is correct. Fixed it with an update of the PR

File: Classes/IndexQueue/RecordMonitor.php [hunk 2]
```diff
@@ -130,7 +130,11 @@ public function processDatamap_afterDatabaseOperations(
...
+            } else {
+                return;
```
[the remark this answers is not in the document]
Comment (magicHatOfTYPO3) [reply Classes/IndexQueue/RecordMonitor.php]: you're right; also fixed this.

**Comment by aschmutt** [comment]
I had the same issue on TYPO3 v11 and PHP8 and can confirm the patch works. 
Merging into master would be great, as I can't create new items in my extension currently.

**Review by dkd-kaehm — APPROVED** [review]
:+1:
But waiting for queued tasks are done, especially in EXT:solrfal.

**Pull request merged**